Implement the Announcement details page

hamo225 opened the pull request:
RISDEV-7045
RISDEV-7046

hamo225 committed:
feat: add AnnouncementDetails view

- includes component for showing metadata

RISDEV-7045
RISDEV-7046

hamo225 committed:
tests: adds unit test for the RisAnnouncementDetails

RISDEV-7045
RISDEV-7046

hamo225 committed:
feat(announcement details):

- adds announcement type
- adds service function for fetching a specific announcement
- updates announcement view to use new service
- removes unneeded e2e test

RISDEV-7045
RISDEV-7046

hamo225 committed:
feat(announcement details view):

- updates announcement type to match response
- adds unit test for new service
- changes url to german (verkundungen)
- updates announcement view new names
- replace amending law with announcement in breadcrumbs

RISDEV-7045
RISDEV-7046

hamo225 committed:
feat(announcement details):

- reverts change to german url name
- reverts extending the norms type to match that of announcements.

RISDEV-7045
RISDEV-7046

hamo225 committed:
feat(announcement details):

 - updates announcement type properties
 - uses correct normExpressionEli when making requests to get an announcement
 - updates routing to english route from home page
 - adds e2e tests
 - adds date formatting
 - change `importTimestamp` to `importedAt` in API schema

RISDEV-7045
RISDEV-7046

hamo225 committed:
tests(e2e): Fix naming of selector

RISDEV-7045
RISDEV-7046

hamo225 committed:
tests(e2e): Fix position of section

RISDEV-7045
RISDEV-7046

hamo225 committed:
tests(a11y): Fixes a11y failings

RISDEV-7045
RISDEV-7046

hamo225 committed:
tests(e2e): Adjust e2e after a11y changes

RISDEV-7045
RISDEV-7046

hamo225 committed:
tests(e2e): Adjust e2e

RISDEV-7045
RISDEV-7046

andreasphil commented on frontend/src/components/controls/RisHeader.vue:
```diff
@@ -219,7 +219,7 @@ export function useHeaderContext() {
   >
     <section class="flex items-center">
-      <Breadcrumb :model="allBreadcrumbs">
+      <Breadcrumb :model="allBreadcrumbs" aria-label="Breadcrumb-Navigation">
```
todo: as discussed, this should not be needed and can be removed

andreasphil commented on frontend/src/components/RisLawPreview.vue:
```diff
@@ -41,13 +41,19 @@ const props = withDefaults(
      * Should the default styling for norms be applied? Default: true
      */
     styled?: boolean
+
```
todo: as discussed, this should not be needed and can be removed. Use `:arrow-focus="false"` instead

hamo225 committed:
fix(a11y): Remove redundant aria-label from RisLaPreview component

Removed because setting :arrow-focus="false" changes the component's role so it no longer functions as a textbox. Components only require an accessible name (via aria-label) when they have an interactive role like textbox, so disabling that behavior eliminates the need for the label.

RISDEV-7045
RISDEV-7046

hamo225 committed:
fix(a11y): Fix unique landmark requirement for nav components

- Removed aria-label from Breadcrumb component
- Added aria-labelledby to RisNavbar and header to ensure unique landmarks

RISDEV-7045
RISDEV-7046

hamo225 committed:
test(e2e): Update e2e test to use fixed importAt date

- API interception to set a
importedAt timestamp

RISDEV-7045
RISDEV-7046
